colossus.Dockerfile

```diff
@@ -1,41 +1,33 @@
-FROM --platform=linux/x86-64 node:14 as builder
+FROM --platform=linux/x86-64 node:18 as builder
 
 WORKDIR /joystream
-COPY . /joystream
+COPY ./chain-metadata.json chain-metadata.json
+COPY ./types types
+COPY ./metadata-protobuf metadata-protobuf
+COPY ./devops/eslint-config ./devops/eslint-config
+COPY ./devops/prettier-config ./devops/prettier-config
+COPY ./storage-node storage-node
+COPY ./yarn.lock yarn.lock
+COPY ./package.json package.json
+COPY ./tsconfig.json ./tsconfig.json
 
 RUN yarn --frozen-lockfile
```

Comment:
The way `Dockerfile` is structured, the npm dependencies will be installed even if only the code changed and, no new dependencies were added in the `package.json` files (in short all the subsequent layers needs to be build again after `COPY ./storage-node storage-node` even if there was only one liner change in storage-node package). Ideally the following build layers should be placed at the top of the `Dockerfile`, 

```
COPY ./yarn.lock yarn.lock
COPY ./types/package.json types/package.json
COPY ./metadata-protobuf/package.json metadata-protobuf/package.json
COPY ./storage-node/package.json storage-node/package.json
COPY ./package.json package.json
RUN yarn --frozen-lockfile
```
so that cached node modules layer can be used whenever there was no change in  `yarn.lock`/`package.json` files, this approach considerably improves the build time of the images.

I think similar docker build optimisations can also be made in `distributor-node.Dockerfile`

Reply:
Great, I'll apply your suggestions and test it out.